lib/browser/crypto.ts

```diff
@@ -13,4 +13,48 @@
  * permissions and limitations under the License.
  */
 
-export const dummy : boolean = false;
+import * as Crypto from "crypto-js";
+import { Hashable } from "../common/crypto";
+
+export class Md5Hash {
+    private hash?: Crypto.WordArray;
+    
+    update(data: Hashable) {
+        this.hash = Crypto.MD5(data.toString(), this.hash ? this.hash.toString() : undefined);
+    }
+
+    digest(truncate_to?: number): DataView {
```

Comment:
I have to confess I like "finalize" better than "digest" both for consistency as well as action vs. noun.

Reply:
Now is the time, I can fix it!